### PR TITLE
Fix track offset calculation when the track range direction is STOP_TO_START

### LINKED ISSUE #8306: Failed to add more than one power restriction
### What happened?

https://github.com/user-attachments/assets/ee22e0bd-1b4e-4d9e-b75a-f2975be83038

### What did you expect to happen?

Can add the power restriction successfully 

### How can we reproduce it (as minimally and precisely as possible)?

1. Edit a train 
2. Add more than one power restriction 
3. Click on "Edit train"
https://rec-osrd.reseau.sncf.fr/operational-studies/projects/39/studies/142/scenarios/264

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

0a34296

### PR DESCRIPTION
**Pull request by achrafmohye**
Closes #8306

### CODE REVIEW

**Comment by emersion** [comment]
Ref https://github.com/OpenRailAssociation/osrd/pull/8218#discussion_r1698473643 which has a discussion about this.

File: front/src/modules/powerRestriction/helpers/__tests__/createPathStep.spec.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ describe('findTrackOffset', () => {
       trackRanges
     );
 
-    expect(result).toEqual({ track: 'track_2', offset: 1050 });
+    expect(result).toEqual({ track: 'track_2', offset: 50 });
```
Comment (clarani) [inline front/src/modules/powerRestriction/helpers/__tests__/createPathStep.spec.ts]: You need to change the input, for the track_2, the begin must be 60 and the end 1060 (as we have seen that the begin < end in the backend response)

**Review by clarani — CHANGES_REQUESTED** [review]

File: front/src/modules/powerRestriction/helpers/__tests__/createPathStep.spec.ts (hunk 1 above)
Comment (achrafmohye) [reply front/src/modules/powerRestriction/helpers/__tests__/createPathStep.spec.ts]: Fixed :)

**Review by clarani — APPROVED** [review]
LGTM :white_check_mark:

**Review by RomainValls — APPROVED** [review]
LGTM

**Pull request merged**